derive_client/data_types/models.py

```diff
@@ -151,6 +221,44 @@ def target_chain(self) -> ChainID:
         return ChainID(self.target_w3.eth.chain_id)
 
 
+@dataclass
+class BridgeTxDetails:
+    contract: Address
+    method: str
+    kwargs: dict[str, Any]
+    tx: dict[str, Any]
+    signed_tx: PSignedTransaction
+
+    @property
+    def tx_hash(self) -> str:
+        """Pre-computed transaction hash."""
+        return self.signed_tx.hash.to_0x_hex
+
+    @property
+    def nonce(self) -> str:
+        """Transaction nonce."""
+        return self.tx["nonce"]
+
+
+@dataclass
+class PreparedBridgeTx:
+    currency: Currency
+    bridge: BridgeType
+    source_chain: ChainID
+    target_chain: ChainID
+    tx_details: BridgeTxDetails
+
+    @property
+    def tx_hash(self) -> str:
+        """Pre-computed transaction hash."""
+        return self.tx_details.tx_hash
+
+    @property
+    def nonce(self) -> str:
+        """Transaction nonce."""
+        return self.tx_details.nonce
```

Comment:
provides user (if they had access to `BridgeClient`s `.prepare_deposit` and `prepare_withdrawal`) the ability to modify and replace their original transaction